dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/jaxrspropagation/JaxRsEndpoints.java

```diff
@@ -178,20 +179,22 @@ public Response getJax(@Context UriInfo uriInfo) {
         LOGGER.info(">>> getJax");
         assertNotNull(Span.current());
 
-        Baggage.builder().put("foo", "bar").build().makeCurrent();
-        Baggage baggage = Baggage.current();
-        assertEquals("bar", baggage.getEntryValue("foo"));
-
-        String url = new String(uriInfo.getAbsolutePath().toString());
-        url = url.replace("jaxrsclient", "jaxrstwo"); //The jaxrsclient will use the URL as given so it needs the final part to be provided.
+        try (Scope s = Baggage.builder().put("foo", "bar").build().makeCurrent()) {
+            Baggage baggage = Baggage.current();
+            assertEquals("bar", baggage.getEntryValue("foo"));
 
-        String result = client.target(url)
-                        .request(MediaType.TEXT_PLAIN)
-                        .get(String.class);
-        assertEquals(TEST_PASSED, result);
+            String url = new String(uriInfo.getAbsolutePath().toString());
+            url = url.replace("jaxrsclient", "jaxrstwo"); //The jaxrsclient will use the URL as given so it needs the final part to be provided.
 
-        LOGGER.info("<<< getJax");
-        return Response.ok(Span.current().getSpanContext().getTraceId()).build();
+            String result = client.target(url)
+                            .request(MediaType.TEXT_PLAIN)
+                            .get(String.class);
+            assertEquals(TEST_PASSED, result);
+            s.close();
+        } finally {
+            LOGGER.info("<<< getJax");
+            return Response.ok(Span.current().getSpanContext().getTraceId()).build();
+        }
```

Comment:
The return statement here must not go in the `finally` block, it should be the last statement in the `try` block.

I'm not entirely sure what this does, but I _think_ that calling `return` in a `finally` block will cause the method to return normally and any exception to be discarded.

In this case, that would mean any `AssertionError` raised by one of the `assert` methods would be lost and the test would pass.

Putting the `LOGGER.info` line in the finally block is ok, I don't really have a preference there.